api/parts/jobs/handle.js

```diff
@@ -14,11 +14,12 @@ class Handle {
     /** constructor **/
     constructor() {
         log.i('Starting job handle in %d', process.pid);
-        this.db = manager.singleDefaultConnection();
-        this.classes = {}; // {'job name': Constructor}
-        this.files = {}; // {'ping': '/usr/local/countly/api/jobs/ping.js'}
-        scan(this.db, this.files, this.classes);
-
+        manager.singleDefaultConnection().then((db) => {
```

Comment:
Why do we still need `manager.singleDefaultConnection`?

Reply:
lesser pool size, I guess
but why do jobs need 2 separate db connections? :D

Reply:
My point is that in your pull request db connectivity stuff should be centralised and no other piece of code should be able to create a separate connection by your wonderful design =)

As for the jobs, I guess you talk about `manager.js` & `handler.js` which just live in separate processes!

Anyway, I'm good with merging.

Reply:
yes, they should be central, I was just not aware of exact use cases there, so decided to leave that up to you :)